xml-parsing/src/test/java/org/sonarsource/analyzer/commons/xml/checks/SimpleXPathBasedCheckTest.java

```diff
@@ -0,0 +1,170 @@
+import java.nio.file.Files;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.xml.xpath.XPathExpression;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+import org.sonarsource.analyzer.commons.xml.XmlFile;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SimpleXPathBasedCheckTest {
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Test(expected = IllegalStateException.class)
+  public void test_invalid_xpath_compilation_throws_exception() throws Exception {
+    new SimpleXPathBasedCheck() {
+
+      XPathExpression failing = getXPathExpression("boolean(a");
+
+      @Override
+      protected void scanFile(XmlFile file) {
+        // do nothing
+      }
+    };
+  }
+
+  @Test
+  public void test_file_making_xpath_evaluation_fail_log_issues() throws Exception {
+    XmlFile xmlFile = getXmlFile("src/test/resources/checks/SimpleXPathBasedCheck/xPathFailure.xml");
+
+    SimpleXPathBasedCheck check = new SimpleXPathBasedCheck() {
+      public XPathExpression comments = getXPathExpression("//comment()");
+
+      @Override
+      protected void scanFile(XmlFile file) {
+        NodeList result = evaluate(comments, file.getDocument());
+        assertThat(result).isNull();
+      }
+    };
+
+    logTester.clear();
+    logTester.setLevel(LoggerLevel.INFO);
+    check.scanFile(null, RuleKey.of("tst", "failingXpath"), xmlFile);
+    assertThat(logTester.logs()).isEmpty();
+
+    logTester.clear();
+    logTester.setLevel(LoggerLevel.DEBUG);
+    check.scanFile(null, RuleKey.of("tst", "failingXpath"), xmlFile);
+    assertThat(logTester.logs()).isNotEmpty();
+    List<String> debugs = logTester.logs(LoggerLevel.DEBUG);
+    assertThat(debugs).hasSize(1);
+    assertThat(debugs.get(0)).startsWith("[tst:failingXpath] Unable to evaluate XPath expression on file ");
+    List<String> errors = logTester.logs(LoggerLevel.ERROR);
+    assertThat(errors).hasSize(1);
+    assertThat(errors.get(0)).startsWith("[tst:failingXpath] XPath exception:");
+  }
+
+  @Test
+  public void test_nodeList() throws Exception {
+    XmlFile xmlFile = getXmlFile("src/test/resources/checks/SimpleXPathBasedCheck/simple.xml");
+
+    SimpleXPathBasedCheck check = new SimpleXPathBasedCheck() {
+      XPathExpression bs = getXPathExpression("//b");
+      XPathExpression ds = getXPathExpression("//d");
+
+      @Override
+      protected void scanFile(XmlFile file) {
+        Document node = file.getDocument();
+
+        NodeList result = evaluate(bs, node);
+        assertThat(result).isNotNull();
```

Comment:
is there a way to assert presence of assertions in the test? I think it's useful when assert is inside nested call like here.